common/src/chain/block/block_v1.rs

```diff
@@ -4,17 +4,17 @@ use crate::chain::transaction::Transaction;
 use crate::chain::ChainConfig;
 use crate::primitives::id;
 use crate::primitives::id::Idable;
-use crate::primitives::{Id, H256};
+use crate::primitives::{Id, Version, H256};
 
 use serialization::{Decode, Encode};
 
 use super::consensus_data::BlockRewardTransactable;
-use super::BlockVersion;
 
-#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Encode, Decode)]
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Encode, Decode, serialization::Tagged)]
 pub struct BlockHeader {
+    pub(super) version: Version<1>,
     #[codec(compact)]
-    pub(super) block_version: u32,
+    pub(super) _reserved: u32,
```

Comment:
Could you please explain how this could help?

Reply:
I just changed the original `block_version` to something else so that the encoding stays the same and tests that hard-code block binary data don't break without updating them for now. Can be removed later.

Reply:
Removed this